apps/filebrowser/src/filebrowser/tasks.py

```diff
@@ -121,63 +136,105 @@ def document_cleanup_task(**kwargs):
-  default_cleanup_threshold = 90
-  cleanup_threshold = kwargs.get('cleanup_threshold', default_cleanup_threshold)
+  cleanup_threshold = kwargs.get('cleanup_threshold', TASK_SERVER_V2.DISK_USAGE_CLEANUP_THRESHOLD)
   username = kwargs.get("username", "celery_scheduler")
-  now = timezone.now()
+  now = timezone.now().astimezone(pytz.timezone(TIME_ZONE))
   kwargs = {
     "username": username,
     "task_name": "tmp_cleanup",
     "task_id": task_id,
     "parameters": cleanup_threshold,
     "progress": "0%",
-    "task_start": now.strftime("%Y-%m-%dT%H:%M:%S")
+    "task_start": now.strftime("%Y-%m-%dT%H:%M:%S.%f"),
   }
 
+  check_disk_usage_and_clean_task.update_state(task_id=task_id, state=states.STARTED, meta=kwargs)
+
+  def delete_old_files(directory, current_time, time_delta):
+    for root, dirs, files in os.walk(directory):
+      for file in files:
+        file_path = os.path.join(root, file)
+        try:
+          file_modified_time = datetime.fromtimestamp(os.path.getmtime(file_path))
+          file_modified_time = timezone.make_aware(file_modified_time, timezone.get_default_timezone())
+          if file_modified_time < current_time - time_delta:
+            if os.path.isfile(file_path) or os.path.islink(file_path):
+              os.unlink(file_path)
+              LOG.info(f"Deleted file {file_path}")
+        except PermissionError:
+          LOG.warning(f"Permission denied: unable to delete {file_path}")
+        except Exception as err:
+          kwargs["task_end"] = timezone.now().astimezone(pytz.timezone(TIME_ZONE)).strftime("%Y-%m-%dT%H:%M:%S.%f")
+          check_disk_usage_and_clean_task.update_state(task_id=task_id, state=states.FAILURE, meta=kwargs)
+          raise Exception(f"Failed to delete {file_path}. Reason: {err}")
+
   disk_usage = psutil.disk_usage('/')
   if disk_usage.percent >= int(cleanup_threshold):
     LOG.info(f"Disk usage is above {cleanup_threshold}%, cleaning up /tmp directory...")
     tmp_dir = '/tmp'
-    for filename in os.listdir(tmp_dir):
-      file_path = os.path.join(tmp_dir, filename)
-      try:
-        file_modified_time = datetime.fromtimestamp(os.path.getmtime(file_path))
-        # Make file_modified_time timezone-aware
-        file_modified_time = timezone.make_aware(file_modified_time, timezone.get_default_timezone())
-        if file_modified_time < now - timedelta(minutes=15):
-          if os.path.isfile(file_path) or os.path.islink(file_path):
-            os.unlink(file_path)
-          elif os.path.isdir(file_path):
-            shutil.rmtree(file_path)
-          LOG.info(f"Deleted {file_path}")
-      except PermissionError:
-        LOG.warning(f"Permission denied: unable to delete {file_path}")
-      except Exception as err:
-        check_disk_usage_and_clean_task.update_state(task_id=task_id, state='FAILURE', meta=kwargs)
-        raise Exception(f"Failed to delete {file_path}. Reason: {err}")
-
+    delete_old_files(tmp_dir, now, timedelta(minutes=60))
```

Comment:
why hard coded timedelta(minutes=60) parameter to 60 minutes. This should be configurable parameter in hue.ini

Reply:
Made it as a configurable parameter in hue.ini `DISK_USAGE_AND_CLEAN_TASK_TIME_DELTA`